### PR TITLE
pygmt.surface: Add aliases for "C", "L", "M", and "T" 

### LINKED ISSUE #2222: [pygmt.surface] tension_factor
### Description of the desired feature

Hello, 
I want to request the feature about surface-tension_factor (-T)
Due to this problem, we need to transfer pygmt to GMT6.2.0 now. 
Hoping the feature in GMT can implement in pygmt in the future.

Thanks! 

### Are you willing to help implement and maintain this feature?

Maybe

### PR DESCRIPTION
**Pull request by JingHuiTong**
**Description of proposed changes**
adding T=''tensor in surface.py

Fixes #2222

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by yvonnefroehlich** [comment]
Welcome 🙂 and great that you started working on this issue @JingHuiTong!

Please add the related issue number #2222 under `Fixes #` by editing your initial comment. Then this issue is automatically closed when this PR is merged.

Beside adding an alias for `T` to the alias list, we also need to add some documentation (docstrings) to explain the usage of this parameter. You may want to orientate yourself on the upstream GMT documentation at https://docs.generic-mapping-tools.org/dev/surface.html#t.
https://github.com/GenericMappingTools/pygmt/blob/8854774eedd2da092cbf2ee623b159ce8310b40a/pygmt/src/surface.py#L55-L71

One tipp for further contributions, which are highly welcome :rocket:: It is better to create a feature branch from the main branch of your personal fork, instead of directly pushing to the main branch.

File: pygmt/src/surface.py
Comment (yvonnefroehlich) [inline pygmt/src/surface.py:23]: I am unsure, whether `tensor` is a good alias for the `T` flag of `surface`. To be consistent with the upstream GMT documentation (https://docs.generic-mapping-tools.org/dev/surface.html#t), I feel `tension` (as suggested by @weiji14) or `tensionfactor` (no underscore because of two bridging consonants) makes for sense.

Comment (weiji14) [reply pygmt/src/surface.py:23]: Let's use `tension`.

```suggestion
    T="tension",
```

Comment (JingHuiTong) [reply pygmt/src/surface.py:23]: OK! I will revise it.

**Comment by JingHuiTong** [comment]
May I add more alias -L -M -C?  (Inspire by #1660)

**Comment by weiji14** [comment]
> May I add more alias -L -M -C?  (Inspire by #1660)

Sure! But you'll need to document each one

File: pygmt/src/surface.py
Comment (weiji14) [inline pygmt/src/surface.py:25]: Change to alphabetical order.
```suggestion
    L="limit",
    M="max_radius",
    R="region",
    G="outgrid",
```

File: pygmt/src/surface.py
Comment (weiji14) [inline pygmt/src/surface.py:75]:
```suggestion
    convergence : float
```

File: pygmt/src/surface.py
Comment (weiji14) [inline pygmt/src/surface.py:84]:
```suggestion
    max_radius : str
```

File: pygmt/src/surface.py
Comment (weiji14) [inline pygmt/src/surface.py:95]:
```suggestion
    tension : float or str
        [**b**\|\ **i**].
```

File: pygmt/src/surface.py
Comment (weiji14) [inline pygmt/src/surface.py:86]:
```suggestion
    max_radius : int or str
```

File: pygmt/src/surface.py
Comment (weiji14) [inline pygmt/src/surface.py:81]: This is how you can add $10^{-4}$ to the docstring.
```suggestion
        of rms deviation by appending %. [Default is scaled to :math:`10^{{-4}}` of the 
```

File: pygmt/src/surface.py
Comment (weiji14) [inline pygmt/src/surface.py:94]:
```suggestion
    lower : float or str
```

File: pygmt/src/surface.py
Comment (weiji14) [inline pygmt/src/surface.py:100]:
```suggestion
    upper : float or str
```

File: pygmt/src/surface.py
Comment (weiji14) [inline pygmt/src/surface.py:106]:
```suggestion
    tension : float or str
```

**Review by weiji14 — COMMENTED** [review]
To fix the 'Style Checks', you can either write `/format` as a comment to have the automated bot format things automatically on GitHub, or do it manually as I mentioned just now in the video call :wink: Let us know if you have any other questions!

**Comment by weiji14** [comment]
Sorry, I meant /format in a GitHub comment, not git commit comment 😅

**Comment by JingHuiTong** [comment]
Ha! I know it after trying.

File: pygmt/src/surface.py
Comment (yvonnefroehlich) [inline pygmt/src/surface.py:25]: Based on this convention in the Contributors Guide (please see section [Code style](https://www.pygmt.org/dev/contributing.html#code-style)), we should use `maxradius` (no underscore) instead of `max_radius`:
> When wrapping a new alias, use an underscore to separate words bridged by vowels (aeiou), such as no_skip and z_only. Do not use an underscore to separate words bridged only by consonants, such as distcalc, and crossprofile. This convention is not applied by the code checking tools, but the PyGMT maintainers will comment on any pull requests as needed.
```suggestion
    M="maxradius",
```

File: pygmt/src/surface.py
Comment (yvonnefroehlich) [inline pygmt/src/surface.py:86]:
```suggestion
    maxradius : int or str
```

File: pygmt/src/surface.py
Comment (yvonnefroehlich) [inline pygmt/src/surface.py:94]: In line 89 the link to upstream GMT documentation to "Units" (https://docs.generic-mapping-tools.org/latest/surface.html#units) is missing (or do we want to remove the term within the brackets?).

```suggestion
        [Default is no masking]. Append a distance unit (see Units) if needed.
        One can also select the nodes to mask by using the *n_cells*\ **c**
        form. Here *n_cells* means the number of cells around the node
        controlled by a data point. As an example ``"0c"`` means that only
        the cell where the point lies is filled, ``"1c"`` keeps one cell
        beyond that (i.e. makes a 3x3 square neighborhood), and so on.
```

File: pygmt/src/surface.py
Comment (yvonnefroehlich) [inline pygmt/src/surface.py:116]:
```suggestion
        (where it tends to flatten the solution approaching the edges). Add 
        **i**\ *tension* to set interior tension, and **b**\ *tension* to 
        set boundary tension. If you do not prepend **i** or **b**, both 
```

File: pygmt/src/surface.py
Comment (yvonnefroehlich) [inline pygmt/src/surface.py:118]:
```suggestion
        will be set to the same value. [Default is 0 for both and gives 
        minimum curvature solution.]
```

File: pygmt/src/surface.py
Comment (michaelgrund) [inline pygmt/src/surface.py:91]:
```suggestion
        Here *n_cells* means the number of cells around the node is controlled 
```

**Comment by JingHuiTong** [comment]
I think it's almost done. Please review it. 
I am truly grateful for your assistance

File: pygmt/src/surface.py
Comment (weiji14) [inline pygmt/src/surface.py:76]: Can also be string if appending the `%` sign to the number.
```suggestion
    convergence : float or str
```

**Review by weiji14 — COMMENTED** [review]
/format

File: pygmt/src/surface.py
Comment (yvonnefroehlich) [inline pygmt/src/surface.py:82]:
```suggestion
        give limit in percentage of rms deviation by appending %. [Default
        is scaled to :math:`10^{{-4}}` of the root-mean-square deviation of
        the data from a best-fit (least-squares) plane.] This is the final
```

File: pygmt/src/surface.py
Comment (weiji14) [inline pygmt/src/surface.py:27]: To let people know what tension values to use, could you change L50 to the following:

```python
    indicates the Laplacian operator. Here, :math:`t = 0` gives the "minimum
    curvature" solution. Minimum curvature can cause undesired oscillations and
    false local maxima or minima (See Smith and Wessel, 1990), and you may wish
    to use :math:`t > 0` to suppress these effects. Experience suggests
    :math:`t \sim 0.25` usually looks good for potential field data and
    :math:`t` should be larger (:math:`t \sim 0.35`) for steep topography data.
    :math:`t = 1` gives a harmonic surface (no maxima or minima are possible
    except at control data points). It is recommended that the user preprocess
    the data with :class:`pygmt.blockmean`, :class:`pygmt.blockmedian`, or
    :class:`pygmt.blockmode` to avoid spatial aliasing and eliminate redundant
    data. You may impose lower and/or upper bounds on the solution. These may
    be entered in the form of a fixed value, a grid with values, or simply be
    the minimum/maximum input data values. Natural boundary conditions are
    applied at the edges, except for geographic data with 360-degree range
    where we apply periodic boundary conditions in the longitude direction.
```

This is copied from https://docs.generic-mapping-tools.org/6.4/surface.html and modified to have proper LaTeX math.

**Review by weiji14 — APPROVED** [review]
Hi @JingHuiTong, hope you had a nice holiday break! Just one last suggestion, otherwise everything looks good to me!

Will wait for one or two other reviewers to approve before we merge this in, just to double check :wink:

File: pygmt/src/surface.py
Comment (seisman) [inline pygmt/src/surface.py:79]:
```suggestion
        ``convergence``. (Units same as data z units). Alternatively,
```

File: pygmt/src/surface.py
Comment (seisman) [inline pygmt/src/surface.py:81]:
```suggestion
        give limit in percentage of root-mean-square (rms) deviation by appending %. [Default
        is scaled to :math:`10^{{-4}}` of the rms deviation of
```

File: pygmt/src/surface.py
Comment (seisman) [inline pygmt/src/surface.py:88]:
```suggestion
        farther than ``maxradius`` away from a data constraint are set to NaN
```

File: pygmt/src/surface.py
Comment (seisman) [inline pygmt/src/surface.py:97]:
```suggestion
        Optional. Impose limits on the output solution. Parameter ``lower``
```

File: pygmt/src/surface.py
Comment (seisman) [inline pygmt/src/surface.py:98]:
```suggestion
        sets the lower bound. ``lower`` can be the name of a grid file with
```

File: pygmt/src/surface.py
Comment (seisman) [inline pygmt/src/surface.py:104]:
```suggestion
        Optional. Impose limits on the output solution. Parameter ``upper``
```

File: pygmt/src/surface.py
Comment (JingHuiTong) [reply pygmt/src/surface.py:27]: No problem. Let me try!  BTW, Happy Chinese New Year! :)

Comment (weiji14) [reply pygmt/src/surface.py:27]: Happy Chinese New Year :firecracker: to you too (even though it's past 元宵 already, haha).

I see that you're encountering some merge conflicts, do you need help with that?

File: examples/gallery/images/grdgradient_shading.py
Comment (michaelgrund) [inline examples/gallery/images/grdgradient_shading.py:12]:
```suggestion
    for i, rad in enumerate(['210/0','210/45', '210/90']):
```
That's not used anymore, right?

**Review by michaelgrund — APPROVED** [review]
Looks good except the minor comment and the merge conflicts.

File: examples/gallery/images/grdgradient_shading.py
Comment (JingHuiTong) [reply examples/gallery/images/grdgradient_shading.py:12]: I'm trying to fix this conflict. Thanks to your remind!

File: pygmt/src/surface.py
Comment (JingHuiTong) [reply pygmt/src/surface.py:27]: I think I solved this conflicts. Can you help me check this?

File: pygmt/src/surface.py
Comment (weiji14) [inline pygmt/src/surface.py:51]: Small detail, but we mostly use neutral (vertical) quotation marks (see https://www.cl.cam.ac.uk/~mgk25/ucs/quotes.html) instead of left/right quotation marks. I know this can be a bit challenging on some keyboards :slightly_smiling_face:

```suggestion
    "minimum curvature" solution. Minimum curvature can cause undesired
```

File: pygmt/src/surface.py
Comment (weiji14) [reply pygmt/src/surface.py:27]: Yep, the extra file seems to be deleted now, and the test and docs build at https://pygmt-dev--2321.org.readthedocs.build/en/2321/api/generated/pygmt.surface.html are looking ok!

**Review by seisman — APPROVED** [review]

File: pygmt/src/surface.py
Comment (yvonnefroehlich) [inline pygmt/src/surface.py:53]:
```suggestion
    oscillations and false local maxima or minima (see Smith and Wessel,
    1990), and you may wish to use :math:`t > 0` to suppress these effects.
```

File: pygmt/src/surface.py
Comment (yvonnefroehlich) [inline pygmt/src/surface.py:60]:
```suggestion
    minima are possible except at control data points). It is recommended that
    the user preprocess the data with :func:`pygmt.blockmean`,
    :func:`pygmt.blockmedian`, or :func:`pygmt.blockmode` to avoid spatial
    aliasing and eliminate redundant data. You may impose lower and/or upper
```

File: pygmt/src/surface.py
Comment (yvonnefroehlich) [inline pygmt/src/surface.py:97]:
```suggestion
        appending %. [Default is scaled to :math:`10^{{-4}}` of the rms
        deviation of the data from a best-fit (least-squares) plane.]
```

File: pygmt/src/surface.py
Comment (yvonnefroehlich) [inline pygmt/src/surface.py:105]:
```suggestion
        [Default is no masking]. Append a distance unit (see
        :gmt-docs:`Units <surface.html#units>`) if needed. One can also
```

**Comment by weiji14** [comment]
Woohoo, thanks again @JingHuiTong! Definitely let us know if you're interested in contributing more stuff, I'm sure you've got plenty of fresh ideas :D

**Pull request merged**
